db/migrate/20230920150053_create_posts.rb

```diff
@@ -0,0 +1,12 @@
+class CreatePosts < ActiveRecord::Migration[7.0]
+  def change
+    create_table :posts do |t|
+      t.references :author, foreign_key: { to_table: :users }
+      t.string :title
+      t.text :text
+      t.integer :comments_counter
+      t.integer :likes_counter
```

Comment:
[OPTIONAL] Please make them default `0` because if the user creates a new post then the `comments_counter` and `likes counter` automatically fill `nil` if you don't make the value default `0` so  kindly add it and please don't skip is an optional requirement :slightly_smiling_face:

Reply:
Dear @aamirkhan2478 ,
Thanks for finding out a probable bug which could cause error in future. I have learnt a lesson that I shouldn't keep any column nil while it should have an int value.

![posts_counter_0](https://github.com/Zafron047/Blog_app_by_Rails/assets/117600135/85705084-b520-4d48-8cd8-9a9192c0607a)


I have made the necessary changes. with [commit](https://github.com/Zafron047/Blog_app_by_Rails/commit/80a495660a01cc381675906cc9b1853b1bd14c6d)

![posts_counter_01](https://github.com/Zafron047/Blog_app_by_Rails/assets/117600135/25099253-9fe7-4cd5-aa56-097a1704d878)


Thanks.